src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -804,6 +804,49 @@ public override SecurityToken ReadToken(XmlReader reader, TokenValidationParamet
             throw LogHelper.LogExceptionMessage(new NotSupportedException(LogMessages.IDX11950));
         }
 
+        /// <summary>
+        /// Converts a string into an instance of <see cref="SamlSecurityToken"/>.
+        /// </summary>
+        /// <param name="token">a Saml token as a string.</param>
+        /// <param name="callContext">An opaque context used to store work when working with authentication artifacts.</param>
+        /// <returns>A <see cref="SamlSecurityToken"/></returns>
+        /// <exception cref="ArgumentNullException">If <paramref name="token"/> is null or empty.</exception>
+        /// <exception cref="ArgumentException">If 'token.Length' is greater than <see cref="TokenHandler.MaximumTokenSizeInBytes"/>.</exception>
+        internal virtual ValidationResult<SamlSecurityToken> ReadSamlToken(string token, CallContext callContext)
+        {
+            if (string.IsNullOrEmpty(token))
+                return ValidationError.NullParameter(nameof(token), ValidationError.GetCurrentStackFrame());
+
+            if (token.Length > MaximumTokenSizeInBytes)
+                return new ValidationError(
+                        new MessageDetail(
+                            TokenLogMessages.IDX10209,
+                            LogHelper.MarkAsNonPII(token.Length),
+                            LogHelper.MarkAsNonPII(MaximumTokenSizeInBytes)),
+                        ValidationFailureType.TokenReadingFailed,
+                        typeof(ArgumentException),
```

Comment:
ArgumentOutOfRangeException would be more correct. Since ArgumentOutOfRangeException derives from ArgumentException this would not be breaking for users.